src/o365/spo/commands/tenant/tenant-recyclebinitem-restore.ts

```diff
@@ -0,0 +1,149 @@
+import request from '../../../../request';
+import commands from '../../commands';
+import SpoCommand from '../../../base/SpoCommand';
+import GlobalOptions from '../../../../GlobalOptions';
+import { CommandOption, CommandCancel } from '../../../../Command';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  url: string;
+  wait: boolean;
+}
+
+interface RestSpoOperation {
+  HasTimedout: boolean;
+  IsComplete: boolean;
+  PollingInterval: number;
+}
+
+class SpoTenantRecycleBinItemRestoreCommand extends SpoCommand {
+  private timeout?: NodeJS.Timer;
+
+  public get name(): string {
+    return commands.TENANT_RECYCLEBINITEM_RESTORE;
+  }
+
+  public get description(): string {
+    return 'Restores the specified deleted Site Collection from Site Collection Recycle Bin';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    const wait: boolean = args.options.wait || false;
+    let spoAdminUrl: string;
+
+    this.getSpoAdminUrl(cmd, this.debug)
+    .then((adminUrl: string): Promise<RestSpoOperation> => {
+      spoAdminUrl = adminUrl;
+      return this.restoreDeletedSite(args.options.url, spoAdminUrl, cmd);
+    })
+    .then((response: RestSpoOperation): void => {
+      if (!response.HasTimedout && response.IsComplete) {
+        if (this.verbose) {
+          cmd.log('Site Collection restored');
+        }
+      }
+      else if (wait) {
+        this.waitForRestoreDeletedSite(args.options.url, spoAdminUrl, cmd, response);
+      }
+      else {
+        cmd.log('Site Collection has not been restored');
```

Comment:
This seems like an error that we should process using `cb('Site Collection has not been restored')`. When done this way and used in a script, it will return a non-zero exit code indicating that that command didn't work as expected.

Reply:
Thank you for pointing that out.